### DISCUSSION TITLE
Can you eager load field on an element matched by route?

### DISCUSSION THREAD

**Discussion by a-solaris** [craftcms/cms#14940] [Q&A] [answered]
Eager loading is a must-have in any site, no matter the complexity, but I was wondering how to deal with eager loading, when the element is matched by routing.

Say I have an Entry with multiple element custom fields, often relating more elements with their own relations. In a template matched by a route, the element is supplied via `entry` variable, preventing me from supplying the `with` value (Craft 4).

Is it possible or feasible to tap into this process? Or is the best approach to take take the `id` from the loaded element and query it again, supplying the `with` value?

**Comment by brandonkelly** [comment, marked as answer]
There’s no need to eager-load the matched entry or its immediate relations, since none of those will be n+1 queries. But if its relations have nested relations, you can eager-load those.

```twig
{% set relatedEntries = entry.myEntriesField
  .with('myAssetsField')
  .all() %}

{% for relatedEntry in relatedEntries %}
  {% for asset in relatedEntry.myAssetsField %}
    ...
  {% endfor %}
{% endfor %}
```

Or in Craft 5 with lazy eager-loading:

```twig
{% for relatedEntry in entry.relatedEntries %}
  {% for asset in relatedEntry.myAssetsField.eagerly() %}
    ...
  {% endfor %}
{% endfor %}
```
